agent/ebs/watcher_test.go

```diff
@@ -300,6 +304,104 @@ func TestHandleDuplicateEBSAttachment(t *testing.T) {
 	assert.True(t, ebsAttachment.IsAttached())
 }
 
+func TestStageAll(t *testing.T) {
+	tcs := []struct {
+		name                     string
+		setTaskStateExpectations func(*mock_dockerstate.MockTaskEngineState)
+		setCSIClientExpectations func(*mock_csiclient.MockCSIClient)
+		expectedErrorStubs       []string
+		expectedAttachmentStatus attachment.AttachmentStatus
+	}{
+		{
+			name: "StageAll succeeds with expected EBS CSI Driver fields",
+			setTaskStateExpectations: func(ts *mock_dockerstate.MockTaskEngineState) {
+				ebsResourceAttachment := getEBSResourceAttachment()
+				ts.EXPECT().GetEBSByVolumeId(taskresourcevolume.TestVolumeId).Return(ebsResourceAttachment, true).Times(2)
+			},
+			setCSIClientExpectations: func(mcc *mock_csiclient.MockCSIClient) {
+				mcc.EXPECT().NodeStageVolume(gomock.Any(),
+					taskresourcevolume.TestVolumeId, gomock.Any(),
+					filepath.Join(hostMountDir, taskresourcevolume.TestSourceVolumeHostPath),
+					taskresourcevolume.TestFileSystem, gomock.Any(), gomock.Any(),
+					gomock.Any(), gomock.Any(), gomock.Any()).Return(nil).Times(1)
+			},
+			expectedErrorStubs:       nil,
+			expectedAttachmentStatus: attachment.AttachmentAttached,
+		},
+		{
+			name: "StageAll returns expected error when CSI NodeStage call fails",
+			setTaskStateExpectations: func(ts *mock_dockerstate.MockTaskEngineState) {
+				ebsResourceAttachment := getEBSResourceAttachment()
+				ts.EXPECT().GetEBSByVolumeId(taskresourcevolume.TestVolumeId).Return(ebsResourceAttachment, true).Times(2)
+			},
+			setCSIClientExpectations: func(mcc *mock_csiclient.MockCSIClient) {
+				mcc.EXPECT().NodeStageVolume(gomock.Any(),
+					taskresourcevolume.TestVolumeId, gomock.Any(),
+					filepath.Join(hostMountDir, taskresourcevolume.TestSourceVolumeHostPath),
+					taskresourcevolume.TestFileSystem, gomock.Any(), gomock.Any(),
+					gomock.Any(), gomock.Any(), gomock.Any()).Return(errors.New("test")).Times(1)
+			},
+			expectedErrorStubs:       []string{"Failed to initialize EBS volume ID"},
+			expectedAttachmentStatus: attachment.AttachmentNone,
+		},
+		{
+			name: "StageAll returns expected error when already attached",
```

Comment:
I think there is no error returned in this case? I am curious why we say "returns expected error".

Reply:
correct -- will change the description.  When already attached we just return `nil`.